include/openmc/distribution_spatial.h

```diff
@@ -94,6 +95,29 @@ class SphericalIndependent : public SpatialDistribution {
   Position origin_;    //!< Cartesian coordinates of the sphere center
 };
 
+//==============================================================================
+//! Distribution of points within a mesh
+//==============================================================================
+
+class MeshIndependent : public SpatialDistribution {
+public:
+  explicit MeshIndependent(pugi::xml_node node);
+
+  //! Sample a position from the distribution
+  //! \param seed Pseudorandom number seed pointer
+  //! \return Sampled position
+  Position sample(uint64_t* seed) const;
+
+private:
+  UnstructuredMesh* umesh_ptr_;
+  int32_t mesh_map_idx_;
+  std::string sample_scheme_;
+  double total_strength_;
+  std::vector<double> mesh_CDF_;
+  std::vector<double> mesh_strengths_;
+  int64_t tot_bins_;
```

Comment:
Since the parameters for mesh min are limited to `int23_t`, `tot_bins` can't exceed the max value of that type -- we can use `int32_t` for `tot_bins` as well.